sktime/base/_meta.py

```diff
@@ -584,6 +587,45 @@ def _tagchain_is_linked_set(
         else:
             self.set_tags(**{mid_tag_name: mid_tag_val_not})
 
+    def _steps_permutation(
+        self,
+        steps: List[Tuple[str, BaseEstimator]],
+        permutation: Optional[List[str]] = None,
+    ) -> List[Tuple[str, BaseEstimator]]:
+        # sort steps by permutation
+        if isinstance(permutation, list):
+
+            # check that all permutation are str values
+            if not all(isinstance(item, str) for item in permutation):
+                raise ValueError("permutation must be a list of strings")
+
+            # check that permutation contains same step names as given in steps
+            if not sorted([x[0] for x in steps]) == sorted(permutation):
+                raise ValueError(
+                    f"""Permutations must contain the same step names as the pipeline,
+                    found tuple names {sorted([x[0] for x in steps])} but got
+                    permutation {sorted(permutation)}."""
+                )
+
+            # sort steps
+            steps_permuted = []
+            for p in permutation:
+                for s in steps:
+                    if p == s[0]:
+                        steps_permuted.append(s)
+            steps = steps_permuted.copy()
+
+        elif permutation == "all":
+            steps = _get_all_permutations(steps)
```

Comment:
I suspect, though would be worth testing - that your estimators will behave strangely if steps is now a list of permutations rather than a single permutation. My suspicion is that there probably needs to be more logic to support the "all" permutations case if you wish to include it.